AvroSourceGenerator/AvroUtils.cs

```diff
@@ -8,21 +9,34 @@ namespace AvroSourceGenerator;
 public static class AvroUtils
 {
     private static readonly SchemaNames SchemaNames = new();
-    
-    public static string GenerateSourceCode(string schemaText)
+
+    public static string GenerateSourceCode(
+        ImmutableArray<(string schemaName, string schemaContents)> schemaNamesAndContents)
     {
-        var schema = Schema.Parse(schemaText, SchemaNames);
-        var codeCompileUnit = new CodeGen
+        var stringWriter = new StringWriter(new StringBuilder());
+
+        var codeGen = new CodeGen();
+        foreach (var (_, schemaContents) in schemaNamesAndContents)
         {
-            Schemas =
-            {
-                schema
-            }
-        }.GenerateCode();
+            var schema = Schema.Parse(schemaContents, SchemaNames);
+
+            codeGen.AddSchema(schema);
+        }
+
+        codeGen.GenerateCode();
+
+        var tempDir = Path.Join(Path.GetTempPath(), $"avro_generated_{Guid.NewGuid():N}");
+        Directory.CreateDirectory(tempDir);
+        codeGen.WriteTypes(tempDir, true);
```

Comment:
this is super hacky, but it works. I'm not 100% sure it'll work on build machines (GHA, etc) where file permissions might be more restrictive. I'm yet to test that out